### ISSUE TITLE
Opt-in to Launch Templates even if no features requires it

### ISSUE THREAD

**Issue by stefansundin** [aws/elastic-beanstalk-roadmap#51]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* Please do not leave "+1" or "me too" comments. They generate extra noise for issue followers and do not help prioritize the request.

**Tell us about your request**
I want a configuration option ([like one of these](https://docs.aws.amazon.com/elasticbeanstalk/latest/dg/command-options.html)) to opt-in to launch templates, even if I don't use a feature that requires them. This would make some of my ebextension cloudformation template overrides simpler since the launch template will always be there.

Something like this:
```yaml
option_settings:
  - namespace: aws:elasticbeanstalk:application
    option_name: Use Launch Template
    value: true
```

**Is this request specific to an Elastic Beanstalk [platform](https://docs.aws.amazon.com/elasticbeanstalk/latest/platforms/platforms-supported.html)?**
No.

**Tell us about the problem you're trying to solve. What are you trying to do, and why is it hard?**
Simplify ebextensions that override things in the cloudformation template.

**Are you currently working around this issue?**
I have some things that is commented out by default, with a comment that it can only be enabled if the deployment is using a launch template.

**Additional context**
If it matters, my app is open source and I make it easy to deploy on both Heroku and Elastic Beanstalk. Here's the repo: https://github.com/stefansundin/rssbox

**Comment by pprahlad** [comment]
We are looking into this request. Thank you for this feedback.

**Comment by gagol14** [comment]
Is there any update or somebody have some tip to force ElasticBeanstalk to use LT instead of LC?

**Comment by darin-holloway** [comment]
@gagol14 use IMDSv2, stub a spot config, and/or use gp3 storage volumes

**Comment by stefansundin** [comment]
Starting on October 1, 2024, new AWS accounts will no longer be able to create launch configurations: https://docs.aws.amazon.com/autoscaling/ec2/userguide/launch-configurations.html

I don't know if they've migrated everything to use Launch Templates now, but I think what @darin-holloway said is probably true, enabling IMDSv2 probably opts you in. Looks like IMDSv2 support was released for Elastic Beanstalk a month after I opened this issue: https://aws.amazon.com/about-aws/whats-new/2020/06/aws-elastic-beanstalk-adds-support-for-imdsv2-and-service-linked-role-for-managed-updates/

Anyway, I will just close this issue since elastic beanstalk is a dying service and no one should really start using it in 2024. There are better options out there now.
